packages/odyssey-react/src/components/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,132 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React, { forwardRef, useEffect, useState } from "react";
+import type { ChangeEvent } from "react";
+import { withTheme } from "@okta/odyssey-react-theme";
+import { useOmit, useSharedRef } from "../../utils";
+import { TextInput, TextInputProps } from "../TextInput";
+import { CloseCircleFilledIcon, SearchIcon } from "../Icon";
+import { Button } from "../Button";
+import { theme } from "./SearchInput.theme";
+import styles from "./SearchInput.module.scss";
+
+export interface SearchInputProps
+  extends Omit<
+    TextInputProps,
+    | "type"
+    | "PrefixButton"
+    | "PrefixIcon"
+    | "PrefixText"
+    | "SuffixButton"
+    | "SuffixIcon"
+    | "SuffixText"
+  > {
+  type?: never;
+  PrefixButton?: never;
+  PrefixIcon?: never;
+  PrefixText?: never;
+  SuffixButton?: never;
+  SuffixIcon?: never;
+  SuffixText?: never;
+}
+/**
+ * Search inputs allow users to enter text search criteria
+ */
+export const SearchInput = withTheme(
+  theme,
+  styles
+)(
+  forwardRef<HTMLInputElement, SearchInputProps>((props, ref) => {
+    const { defaultValue, onChange, label, labelHidden, name, value, ...rest } =
+      props;
+
+    const omitProps = useOmit(rest);
+    const internalRef = useSharedRef(ref);
+    const [isControlled, setIsControlled] = useState(
+      typeof value !== "undefined"
+    );
+    const [canShowClearButton, setCanShowClearButton] = useState(
+      isControlled ? !!value : !!defaultValue
+    );
+
+    useEffect(() => {
+      if (typeof value === "undefined") {
+        setIsControlled(false);
+      } else {
+        setIsControlled(true);
+        setCanShowClearButton(!!value);
+      }
+    }, [value]);
+
+    const setFocus = () => {
+      requestAnimationFrame(() => {
+        internalRef.current?.focus();
+      });
+    };
+
+    const onClear = () => {
+      if (internalRef.current) {
+        setFocus();
+        const nativeInputValueSetter = Object.getOwnPropertyDescriptor(
+          window?.HTMLInputElement?.prototype,
+          "value"
+        )?.set;
+        if (nativeInputValueSetter) {
+          nativeInputValueSetter.call(internalRef.current, "");
+          const aChangeEvent = new Event("change", { bubbles: true });
+          internalRef.current.dispatchEvent(aChangeEvent);
+        }
+      }
+    };
+
+    const internalOnChange = (event: ChangeEvent<HTMLInputElement>) => {
+      if (!isControlled) {
+        setCanShowClearButton(!!event.target.value);
+      }
+      onChange?.(event, event.target.value);
+    };
+
+    return (
+      <span className={styles.search}>
+        {/*
```

Comment:
@jeffbelser-okta - I'm having trouble with this exception. The correct syntax to get this to work is supposed to be: 
```
  {/*
     // @ts-expect-error using a type="search" intentionally here */}
    <TextInput {...omitProps} type="search"
        ...
```

But something in the linting rules changes it to this format here which causes the build to fail..

Reply:
Perhaps prettier or eslint rules, I will checkout your branch to see what can be done here...

Reply:
ok, thanks

Reply:
Passing a class here would be an effort to avoid that problem. Search's `.root` and `.clear` affect functionality that's provided by `TextInput` as well as adding otherwise unnecessary HTML. Similarly, we house our `type="search"` styles with `TextInput`, not `SearchInput`.

For a practical example, a feature team wishing to utilize hide/show behavior shouldn't need to build a custom component. Hiding and showing `TextInput` affixes aren't solely the domain of `SearchInput`. Rather, `TextInput` could handle that logic and styling for them.

It's not a blocker for this PR, but this duplication will likely continue to increase as we/teams create other `type`s like `ClickToCopy` or `Number`. For contrast, styles for components like `DatePicker` would be sequestered as they're truly unique.